lib/upgrade_format_task.rb

```diff
@@ -44,7 +44,9 @@ def upgrade(store, name, options = {})
 
     if !errors.empty?
       Machinery.logger.error errors.join("\n")
-      raise Machinery::Errors::UpgradeFailed.new(errors.join("\n"))
+      exception = Machinery::Errors::UpgradeFailed.new(errors.join("\n"))
+      exception.hint_proc = -> { Hint.upgrade_format_force(name: name || "--all") }
```

Comment:
It needs some effort to understand this code and you need to know what the `hint_proc` method is supposed to do. Wouldn't it be easier to not introduce special magic here and use something like:

```
exception = Machinery::Errors::UpgradeFailed.new(errors.join("\n") + Hint.upgrade_format_force(name: name || "--all", to_string: true)
```

where the `to_string` option would get the hint returned as string instead of being directly put out on STDERR.